cmd/rqlite/http/client_test.go

```diff
@@ -0,0 +1,162 @@
+package http
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+)
+
+func TestClient_QueryWhenAllAvailable(t *testing.T) {
+	node1 := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		writer.WriteHeader(http.StatusOK)
+	}))
+	defer node1.Close()
+
+	node2 := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		writer.WriteHeader(http.StatusOK)
+	}))
+	defer node2.Close()
+
+	httpClient := http.DefaultClient
+
+	u1, _ := url.Parse(node1.URL)
+	u2, _ := url.Parse(node2.URL)
+	client := NewClient(httpClient, []string{u1.Host, u2.Host})
+	res, err := client.Query(url.URL{
+		Path: "/",
+	})
+	defer res.Body.Close()
+
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	if client.currentHost != 0 {
+		t.Errorf("expected to only forward requests to the first host")
+	}
+
+	if res.StatusCode != http.StatusOK {
+		t.Errorf("unexpected status code, expected '200' got '%d'", res.StatusCode)
+	}
+}
+
+func TestClient_QueryWhenSomeAreAvailable(t *testing.T) {
+	node1 := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		writer.WriteHeader(http.StatusServiceUnavailable)
+	}))
+	defer node1.Close()
+
+	node2 := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		writer.WriteHeader(http.StatusOK)
+	}))
+	defer node2.Close()
+
+	httpClient := http.DefaultClient
+
+	u1, _ := url.Parse(node1.URL)
+	u2, _ := url.Parse(node2.URL)
+	client := NewClient(httpClient, []string{u1.Host, u2.Host})
+	res, err := client.Query(url.URL{
+		Path: "/",
+	})
+	defer res.Body.Close()
+
+	if err == nil {
+		t.Errorf("expected error")
```

Comment:
Minor -- can you add a little more explanation to this string? Why was an error expected? This will help understand the test better, especially if it fails.